controller/nodesyncer/nodessyncer.go

```diff
@@ -119,3 +124,14 @@ func (n *NodeSyncer) convertNodeToClusterNode(node *v1.Node) *clusterv1.ClusterN
 	}
 	return clusterNode
 }
+
+func (n *NodeSyncer) isClusterRemoving(clusterNode *clusterv1.ClusterNode) (bool, error) {
```

Comment:
So here is the deal :) this method should return error only when k8s api call errors out with non-404 error. When there is an error and 404, return true, nil. When non error and removing timestamp is set, return true, nil. In all other cases, return false, nil.